gnosis/eth/eip712/__init__.py

```diff
@@ -57,6 +57,14 @@ def _encode_field(name, typ, value):
         if value is None:
             raise Exception(f"Missing value for field {name} of type {type}")
 
+        # Accept string bytes
+        if "bytes" in typ and isinstance(value, str):
+            value = bytes.fromhex(value.replace("0x", ""))
+
+        # Accept string uint
```

Comment:
I think we should support any kind of integer, not only `uint`. This is a very misleading behaviour:

`if "int" in typ`